### ISSUE TITLE
Docker running on Azure VM cannot switch to Linux containers.

### ISSUE THREAD

**Issue by Youkko** [docker/for-win#10629]
- [X] I have tried with the latest version of Docker Desktop
  - [X] I have tried disabling enabled experimental features
  - [X] I have uploaded Diagnostics
  - Diagnostics ID: 36673C52-7D76-4417-BE6D-1B474ED97054/20210311184611

### Expected behavior
Docker should switch to Linux containers.

### Actual behavior
Docker crashes when I try to switch to linux containers.

### Information
I installed latest docker desktop on a Windows 10 VM inside Azure. I enabled WSL2 and Hyper-V, and I am able to use it on Windows containers mode. But whenever I try to switch to Linux containers, docker crashes. It says that I must install WSL2, but... I already did.

I tried uninstalling everything - WSL Kernel update, Docker - and disabling Hyper-V, then restarting, then deleting every trace of Docker, then reinstalling and doing all over again, but the problem keeps happening.

I don't know what to do to solve this.

I'm using Windows 10 Pro (Version 1909, build 18363.1377)

Client: Docker Engine - Community
 Cloud integration: 1.0.9
 Version:           20.10.5
 API version:       1.41
 Go version:        go1.13.15
 Git commit:        55c4c88
 Built:             Tue Mar  2 20:14:53 2021
 OS/Arch:           windows/amd64
 Context:           default
 Experimental:      true

Server: Docker Engine - Community
 Engine:
  Version:          20.10.5
  API version:      1.41 (minimum version 1.24)
  Go version:       go1.13.15
  Git commit:       363e9a8
  Built:            Tue Mar  2 20:26:56 2021
  OS/Arch:          windows/amd64
  Experimental:     false

**Comment by stephen-turner** [comment]
What does `wsl -l -v` say?

**Comment by Youkko** [comment]
> What does `wsl -l -v` say?

PS C:\windows\system32> wsl -l -v
Windows Subsystem for Linux has no installed distributions.
Distributions can be installed by visiting the Microsoft Store:
https://aka.ms/wslstore

**Comment by Youkko** [comment]
> It's presumably a problem with enabling virtualisation in a nested virt environment. (Windows containers don't have to run in a VM / WSL 2 environment).

I believe that you are correct about nested virtualization. I studied a lot yesterday, and got to this very same conclusion, specially when I got to Hyper-V Manager and tried to manually start the "DockerDesktopVM": it errors saying "Failed to start the virtual machine 'DockerDesktopVM' because one of the Hyper-V components is not running'.
If that's the case, my hands are tied. That's a company VM, and I don't have the authority or permission level to manage it's settings on Azure. Also, if it is required to have a Linux distro installed from MS Store, my hands are double-tied: I also can't access MS Store because of company policies.
